lib/core/backend/backendImport.ts

```diff
@@ -0,0 +1,174 @@
+    }
+    else if (! isPlainObject(mappings)) {
+      throw assertionError.get('invalid_type', 'mappings', 'object');
+    }
+
+    for (const index of Object.keys(mappings)) {
+      if (! isPlainObject(mappings[index])) {
+        throw assertionError.get('invalid_type', `mappings.${index}`, 'object');
+      }
+      // If some collections have already been defined, only their last mappings will be retained
+      this._application._import.mappings[index] =
+        Object.assign({}, this._application._import.mappings[index], mappings[index]);
+    }
+  }
+
+  /**
+   * Import profiles.
+   *
+   * Example:
+   * {
+   *   profileA: { profile definition },
+   *   profileB: { profile definition },
+   *   ...
+   * }
+   *
+   * @param profiles Object containing profiles and their definition
+   */
+  profiles (profiles: JSONObject) : void {
+    if (this._application.started) {
+      throw runtimeError.get('already_started', 'import');
+    }
+    else if (! isPlainObject(profiles)) {
+      throw assertionError.get('invalid_type', 'profiles', 'object');
+    }
+
+    // If some profiles have already been defined, only their last definition will be retained
+    this._application._import.profiles =
+      Object.assign({}, this._application._import.profiles, profiles);
+  }
+
+  /**
+   * Import roles
+   *
+   * Example:
+   * {
+   *   roleA: { role definition },
+   *   roleB: { role definition },
+   *   ...
+   * }
+   *
+   * @param roles Object containing roles and their definition
+   */
+  roles (roles: JSONObject) : void {
+    if (this._application.started) {
+      throw runtimeError.get('already_started', 'import');
+    }
+    else if (! isPlainObject(roles)) {
+      throw assertionError.get('invalid_type', 'roles', 'object');
+    }
+
+    // If some roles have already been defined, only their last definition will be retained
+    this._application._import.roles =
+      Object.assign({}, this._application._import.roles, roles);
+  }
+
+  /**
+   * Import user mappings.
+   *
+   * Example:
+   * {
+   *   properties: {
+   *     fieldA: { type: 'keyword' },
+   *     fieldB: { type: 'integer' },
+   *     ...
+   *   }
+   * }
+   *
+   * @param mappings User properties
+   */
+  userMappings (mappings: JSONObject) : void {
+    if (this._application.started) {
+      throw runtimeError.get('already_started', 'import');
+    }
+    else if (! isPlainObject(mappings)) {
+      throw assertionError.get('invalid_type', 'mappings', 'object');
+    }
+
+    this._application._import.userMappings = mappings;
+  }
+
+  /**
+   * Import users.
+   *
+   * Example:
+   * {
+   *   kuidA: { user content },
+   *   kuidB: { user content },
+   * }
+   *
+   * @param users Object containing users and their content
+   * @param onExistingUsers Default to `skip`. Strategy to adopt when trying to create an already existing user.
+   */
+  users (users: JSONObject, onExistingUsers?: 'fail' | 'overwrite' | 'skip') : void {
```

Comment:
As said earlier, open should be passed in object for readability and extensibility